datahub-web-react/src/app/entity/user/UserEditProfileModal.tsx

```diff
@@ -0,0 +1,242 @@
+import React, { useEffect, useState } from 'react';
+import { message, Button, Input, Modal, Typography, Form } from 'antd';
+import { useUpdateCorpUserPropertiesMutation } from '../../../graphql/user.generated';
+
+type Props = {
+    visible: boolean;
+    onClose: () => void;
+    onCreate: () => void;
+    data: {
+        name: string | undefined;
+        title: string | undefined;
+        image: string | undefined;
+        team: string | undefined;
+        email: string | undefined;
+        slack: string | undefined;
+        phone: string | undefined;
+        urn: string | undefined;
+    };
+};
+/** Regex Validations */
+export const validUserName = new RegExp('^[a-zA-Z ]*$');
+
+export default function UserEditProfileModal({ visible, onClose, onCreate, data }: Props) {
+    const [updateCorpUserPropertiesMutation] = useUpdateCorpUserPropertiesMutation();
+    const [form] = Form.useForm();
+
+    const [userName, setUserName] = useState<string | undefined>('');
+    const [userTitle, setUserTitle] = useState<string | undefined>('');
+    const [userImageURL, setImageURL] = useState<string | undefined>('');
+    const [userTeam, setUserTeam] = useState<string | undefined>('');
+    const [userEmail, setUserEmail] = useState<string | undefined>('');
+    const [userSlack, setUserSlack] = useState<string | undefined>('');
+    const [userPhoneNumber, setUserPhoneNumber] = useState<string | undefined>('');
+    const [buttonDisabled, setButtonDisabled] = useState(true);
```

Comment:
let's set the default values to be what is provided by the API in the case they already have a value

Reply:
ah nvm- just see that you do this below via useEffect

Reply:
Looking again though, you use `data` to provide initial values to the form below. In that case, if you are using data for default values below, you should be able to just do

```const [userTitle, setUserTitle] = useState<string | undefined>(title);``` and skip the useEffect at line 36

Reply:
Sure. Required changes added.